src/main.rs

```diff
@@ -83,7 +96,81 @@ fn main() {
         Rank::InfoPlusProbability => roget::Rank::InfoPlusProbability,
         Rank::ExpectedInformation => roget::Rank::ExpectedInformation,
     };
-    play(move || solver.build(), args.games);
+    if args.interactive {
+        play_interactive(solver.build());
+    } else {
+        play(move || solver.build(), args.games);
+    }
+}
+
+fn play_interactive(mut guesser: impl Guesser) {
+    let mut history = Vec::with_capacity(6);
+    println!("C: Correct / Green, M: Misplaced / Yellow, W: Wrong / Gray");
+    // Wordle only allows six guesses.
+    for _ in 1..=6 {
+        let guess = guesser.guess(&history);
+        println!("Guess:  {}", guess.to_uppercase());
+        let correctness = {
+            loop {
+                match ask_for_correctness() {
+                    Ok(c) => break c,
+                    Err(e) => println!("{}", e),
+                }
+            }
+        };
+        if correctness == [roget::Correctness::Correct; 5] {
+            println!("The answer was {}", guess.to_uppercase());
+            return;
+        }
+        history.push(roget::Guess {
+            word: Cow::Owned(guess),
+            mask: correctness,
+        });
+    }
+    println!("Game Over, only six guesses are allowed");
+}
+
+fn ask_for_correctness() -> Result<[roget::Correctness; 5], Cow<'static, str>> {
+    print!("Colors: ");
+    std::io::Write::flush(&mut std::io::stdout()).unwrap();
+    let mut answer = String::with_capacity(7);
+    std::io::stdin().read_line(&mut answer).unwrap();
+    let answer = answer
+        .trim()
+        .chars()
+        .filter(|v| !v.is_whitespace())
+        .map(|v| v.to_ascii_uppercase())
+        .collect::<String>();
+    if answer.len() != 5 {
+        Err("The correctness string should be 5 chars long")?;
+    }
+    let parsed = answer
+        .chars()
+        .map(|c| match c {
+            'C' => Ok(roget::Correctness::Correct),
+            'M' => Ok(roget::Correctness::Misplaced),
+            'W' => Ok(roget::Correctness::Wrong),
+            _ => Err(format!("Invalid correctness char: '{c}'")),
+        })
+        .collect::<Vec<_>>();
```

Comment:
```suggestion
        .collect::<Result<Vec<_>, _>>()?;
```
This way you can also get rid of the `if let Some` below.

Reply:
Thanks for the information, I was looking for a way to do exactly that.